File: add-digits/add-digits.py
class Solution:
    def addDigits(self, num: int) -> int:
        ans = 0 
        if num<=9:
            return num 
        while(num!=0):
            ans += num % 10 
            num //= 10 
            if(ans>9 and num==0):
                num = ans 
                ans = 0 
        return ans
        

File: add-digits/add-digits.cpp
class Solution {
public:
    int addDigits(int num) {
        int ans = 0 ;
        if(num<=9)
        return num ;
        while(num!=0){
            ans += num%10 ;
            num /= 10 ;
            if(ans>9 && num==0){
                num = ans ;
                ans = 0;
            }
        }
        return ans;
        
    }
};